airflow/ti_deps/deps/ready_to_reschedule.py

```diff
@@ -52,14 +52,14 @@ def _get_dep_statuses(self, ti, session, dep_context):
                 reason="The task instance is not in State_UP_FOR_RESCHEDULE or NONE state.")
             return
 
-        task_reschedules = TaskReschedule.find_for_task_instance(task_instance=ti)
-        if not task_reschedules:
+        task_reschedule = TaskReschedule.query_for_task_instance(task_instance=ti, descending=True).first()
```

Comment:
```suggestion
        task_reschedule = (
            TaskReschedule.query_for_task_instance(task_instance=ti, descending=True)
            .with_entities(TR.reschedule_date)
            .first()
        )
```

Reply:
Oh does `with_entites` say "just select this one column? Nice!

Reply:
This allows you to change the content of the parameter passed to the query method on an existing object. We already use it in the project. https://github.com/apache/airflow/pull/8729/files

Reply:
Here is doc: https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.with_entities